pkg/flow/internal/controller/node_native_component.go

```diff
@@ -111,11 +111,12 @@ type ComponentNode struct {
 	exports    component.Exports // Evaluated exports for the managed component
 }
 
-var _ BlockNode = (*ComponentNode)(nil)
+var _ NodeWithDependants = (*NativeComponentNode)(nil)
+var _ ComponentNode = (*NativeComponentNode)(nil)
 
 // NewComponentNode creates a new ComponentNode from an initial ast.BlockStmt.
 // The underlying managed component isn't created until Evaluate is called.
-func NewComponentNode(globals ComponentGlobals, reg component.Registration, b *ast.BlockStmt) *ComponentNode {
+func NewComponentNode(globals ComponentGlobals, reg component.Registration, b *ast.BlockStmt) *NativeComponentNode {
```

Comment:
The function name here should be updated to NewNativeComponentNode.

Reply:
good catch, missed it during the renaming